### ISSUE TITLE
Not possible to store dates in DB

### ISSUE THREAD

**Issue by chabad360** [hebcal/hebcal-go#3]
Up until v0.9.11 I was storing the `HDate` directly in the database (as a JSON object). However, now that the fields are private this is no longer possible as the JSON encoding no longer discovers those fields. Would it be possible to add support for the [`encoding/json.Marshaller`](https://pkg.go.dev/encoding/json#Marshaler) and [`Unmarshaller`](https://pkg.go.dev/encoding/json#Unmarshaler) interfaces (or simply make the fields public again)?

**Comment by mjradwin** [comment]
Thanks for the feedback. I'm still new to Golang and have been trying to follow best practices.

We'll look into the JSON APIs and try to get a release out

**Comment by mjradwin** [comment]
The format implemented matches what is currently used on hebcal.com. We have a preference for string month names because in the world of Hebrew dates there can be ambiguity about whether month 1 is Nissan or Tishrei.

Sorry for the compatibility break. We have been releasing this library as 0.9.x because we expected some breaking changes like this.

**Comment by chabad360** [comment]
Fair enough. I'll have to see how I'm going to work around that, but this is a good start.

**Comment by mjradwin** [comment]
Our apologies for the inconvenience.

We're hoping to avoid hassles like this in the future by stabilizing some key interfaces. `hdate` is now its package split out from this monorepo into its own repo/module. We've given it a `v1.0.0` stable version number, and won't make any breaking changes without increasing the major version number.

Instead of importing `github.com/hebcal/hebcal-go/hdate`, you'll now want to import `github.com/hebcal/hdate`
